### PR TITLE
add patterns_05 & multithreading_06 tasks

### PR DESCRIPTION
**Pull request by wilden5**
Applied Singleton and Command patterns to project

### CODE REVIEW

**Comment by wilden5** [comment]
I am added **multithreading_06** task to this PR and refactored patterns

File: consoleApp/src/main/java/com/denis/consoleapp/service/PrintOrderCommand.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    private Store store;
+    private int categoryNumber;
+    private int productNumber;
+
+    public PrintOrderCommand(Store store) {
+        this.store = store;
+    }
+
+    @Override
+    public void execute() {
```
Comment (artemmoroz) [inline consoleApp/src/main/java/com/denis/consoleapp/service/PrintOrderCommand.java]: please think about refactoring of these method(class), cause now it looks like mess. some improvements: 1. move schedulers to separate method/class, 2. take a look at performance, cause you read data N times in cycles, you can optimize it, 3. printiing

File: consoleApp/src/main/java/com/denis/consoleapp/service/PrintTopPriceCommand.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            stringBuilder.append(".");
+            stringBuilder.append(topSortProducts.get(i));
+            stringBuilder.append("\n");
+        }
+        System.out.println("Top5 products from all categories sorted via price DESC: ");
+        System.out.println(stringBuilder);
+    }
+
+    private List<Product> getTopSortProduct() {
+        ArrayList<Product> allProducts = new ArrayList<>();
```
Comment (artemmoroz) [inline consoleApp/src/main/java/com/denis/consoleapp/service/PrintTopPriceCommand.java]: List... = new ArrayList

File: consoleApp/src/main/java/com/denis/consoleapp/service/PrintTopPriceCommand.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        HashMap sortParams = new HashMap<>();
```
Comment (artemmoroz) [inline consoleApp/src/main/java/com/denis/consoleapp/service/PrintTopPriceCommand.java]: Map ....

File: consoleApp/src/main/java/com/denis/consoleapp/service/Switch.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package com.denis.consoleapp.service;
+
+public class Switch {
```
Comment (artemmoroz) [inline consoleApp/src/main/java/com/denis/consoleapp/service/Switch.java]: i did not get the idea of this class, if you want to organize commands - it is not the best solution. please make a deeper look on command pattern or chain of responsibility pattern

File: store/src/main/java/com/denis/store/Store.java [hunk 5]
```diff
@@ -2,26 +2,31 @@
...
-    public Store() {
+    private Store() {
         try {
             this.categoryList = new RandomStorePopulator().getRandomCategory();
         } catch (Exception exception) {
             System.out.println("ERROR: RandomStorePopulator = " + exception.getMessage());
         }
     }
 
+    public static Store getInstance() {
```
Comment (artemmoroz) [inline store/src/main/java/com/denis/store/Store.java]: what for it is singleton? if just to apply this pattern - it is not the best choice, if not - please comment

**Review by artemmoroz — CHANGES_REQUESTED** [review]
in the future, you'd better use different jumps for different tasks

**Review by artemmoroz — CHANGES_REQUESTED** [review]
phone call

File: store/src/main/java/com/denis/store/utility/BaseString.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
+package com.denis.store.utility;
+
+import com.denis.domain.Category;
+import com.denis.domain.Product;
+import com.denis.store.Store;
+
+import java.util.List;
+
+public class BaseString {
```
Comment (artemmoroz) [inline store/src/main/java/com/denis/store/utility/BaseString.java]: very strange class name, may be PrintService, or PrintHelper would be better

File: store/src/main/java/com/denis/store/utility/BaseString.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+    public String printCategories(List<Category> categoryList) {
```
Comment (artemmoroz) [inline store/src/main/java/com/denis/store/utility/BaseString.java]: these methods (this and above) are utilities methods, and could be static, it that case you shhould not create BaseString instance and provide it to other methods.

**Review by artemmoroz — CHANGES_REQUESTED** [review]

**Comment by wilden5** [comment]
Fixed the last issues

**Review by artemmoroz — APPROVED** [review]

**Pull request merged**